presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/RecordingHiveMetastoreModule.java

```diff
@@ -28,24 +32,52 @@
 import static org.weakref.jmx.guice.ExportBinder.newExporter;
 
 public class RecordingHiveMetastoreModule
```

Comment:
Maybe RecordingAwareHiveMetastoreModule. Or OptionalRecordingHiveMetastoreModule. As now it has conditional logic inside.

Reply:
modelled after `CachingHiveMetastoreModule` -- the module is named caching even though caching can be enabled or disabled

also, i am biding with `ForRecordingHiveMetastore` and then installing the `RecordingHiveMetastoreModule`, so `RecordingHiveMetastoreModule` must fulfil the contract even if recording is disabled.